docs/proposals/config_data.md

```diff
@@ -0,0 +1,327 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores data used for the configuration of ``Kubernetes`` components.
```

Comment:
ConfigData is intended to directly address #1553, which will enable dynamic distribution of configuration to applications without coupling them to key-value stores.